streaming/out.go

```diff
@@ -20,22 +20,41 @@ type Crypt4GHWriter struct {
 	header                               headers.Header
 	dataEncryptionParametersHeaderPacket headers.DataEncryptionParametersHeaderPacket
 	buffer                               bytes.Buffer
+	Rands                                *WriterRands
+}
+
+type WriterRands struct {
+	dataKey      [chacha20poly1305.KeySize]byte
+	headerNonces []*[chacha20poly1305.NonceSize]byte
+	bodyNonces   []*[chacha20poly1305.NonceSize]byte
 }
 
 // NewCrypt4GHWriter method constructs streaming.Crypt4GHWriter instance from io.Writer and corresponding keys.
-func NewCrypt4GHWriter(writer io.Writer, writerPrivateKey [chacha20poly1305.KeySize]byte, readerPublicKeyList [][chacha20poly1305.KeySize]byte, dataEditList *headers.DataEditListHeaderPacket) (*Crypt4GHWriter, error) {
+func NewCrypt4GHWriter(
```

Comment:
Could we keep the old interface please (and have this as e.g. `NewCrypt4GHWriterWithRands` or similar)?

Reply:
ok